frontend/src/routes/Governance/discovered/grouping.ts

```diff
@@ -0,0 +1,226 @@
+/* Copyright Contributors to the Open Cluster Management project */
+type TResolveSource = (annotations: { [key: string]: any }, helmReleases: any, channels: any, subscriptions: any) => any
+
+type TGetSourceText = (policySource: any, isExternal: boolean, t: any) => string
+
+type TTransformAnnotation = (anoString: string) => { [key: string]: string }
+
+type TGetStringLabelValue = (labels: string, labelKey: string) => string | undefined
+
+type TParseDiscoveryPolicies = (data: any) => any
+interface ISourceType {
+  type: any //ex: 'Policy' | 'Git' | 'Multiple'
+  parentNs: string
+  parentName: string
+}
+
+export function grouping(): {
+  getPolicySource: (
+    policy: any,
+    helmReleases: any[],
+    channels: any[],
+    subscriptions: any[],
+    resolveSource: TResolveSource,
+    getSourceText: TGetSourceText,
+    getStringLabelValue: TGetStringLabelValue,
+    transformAnnotation: TTransformAnnotation
+  ) => ISourceType
+  createMessage: (
+    data: any[],
+    helmReleases: any[],
+    channels: any[],
+    subscriptions: any[],
+    resolveSource: string,
+    getSourceText: string,
+    getStringLabelValue: string,
+    transformAnnotation: string,
+    parseDiscoveryPolicies: string
+  ) => any[]
+} {
+  const getPolicySource = (
+    policy: any,
+    helmReleases: any[],
+    channels: any[],
+    subscriptions: any[],
+    resolveSource: TResolveSource,
+    getSourceText: TGetSourceText,
+    getStringLabelValue: TGetStringLabelValue,
+    transformAnnotation: TTransformAnnotation
+  ): ISourceType => {
+    const parentPolicy = getStringLabelValue(policy.label, 'policy.open-cluster-management.io/policy')
+    const parentNs = parentPolicy?.split('.')[0] ?? ''
+    const parentName = parentPolicy?.split('.')[1] ?? ''
+
+    if (parentNs && parentName) {
+      return {
+        type: 'Policy',
+        parentNs,
+        parentName,
+      }
+    }
+
+    const isExternal = policy._isExternal
+    let source: string = 'Local'
+    if (isExternal) {
+      const policySource = resolveSource(transformAnnotation(policy.annotations), helmReleases, channels, subscriptions)
+      source = policySource ? getSourceText(policySource, isExternal, (s: string) => s) : 'Managed externally'
+    }
+    // source is string
+    return {
+      type: source,
+      parentNs: '',
+      parentName: '',
+    }
+  }
+
+  enum PolicySeverity {
+    Unknown,
+    Low,
+    Medium,
+    High,
+    Critical,
+  }
+
+  const severityTable = (severity: string): number => {
+    if (!severity) return PolicySeverity.High
```

Comment:
This doesn't block the PR, but let's bring this up in the sync call to see what others think of this as a default severity.